include/swift/SIL/SILValue.h

```diff
@@ -618,6 +616,114 @@ class OwnershipConstraint {
 llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
                               OwnershipConstraint constraint);
 
+/// Categorize all uses in terms of their ownership effect.
+///
+/// Used to verify completeness of the ownership use model and exhaustively
+/// switch over any category of ownership use. Implies ownership constraints and
+/// lifetime constraints.
+enum class OperandOwnership {
+  /// Uses of ownership None. These uses are incompatible with values that have
+  /// ownership but are otherwise not verified.
+  None,
+
+  /// MARK: Uses of any ownership values:
+
+  /// Point-in-time use. Uses the value instantaneously.
+  /// (copy_value, single-instruction apply with @guaranteed argument)
+  InstantaneousUse,
+  // FIXME: The PointerEscape category should be eliminated. All pointer escapes
+  // should be InteriorPointer, guarded by a borrow scope.
+  PointerEscape,
+  /// Bitwise escape. Escapes the nontrivial contents of the value.
+  /// OSSA does not enforce the lifetime of the escaping bits.
+  /// The programmer must explicitly force lifetime extension.
+  /// (ref_to_unowned, unchecked_trivial_bitcast)
+  BitwiseEscape,
+
+  /// MARK: Uses of Owned values:
+
+  /// Borrow. Propagates the owned value within a scope, without consuming it.
+  /// (begin_borrow, begin_apply with @guaranteed argument)
+  Borrow,
+  /// Destroying Consume. Destroys the owned value immediately.
+  /// (store, destroy, @owned, destructure).
+  DestroyingConsume,
+  /// Forwarding Consume. Consumes the owned value indirectly via a move.
+  /// (br, destructure, tuple, struct, cast, switch).
+  ForwardingConsume,
+
+  /// MARK: Uses of Guaranteed values:
+
+  /// Nested Borrow. Propagates the guaranteed value within a nested borrow
+  /// scope, without ending the outer borrow scope, following stack discipline.
+  /// (begin_borrow, begin_apply with @guaranteed).
+  NestedBorrow,
+  /// Interior Pointer. Propagates the guaranteed value within the base's
```

Comment:
I don't have good verbage... but this should be clear that an Interior Pointer is meant for instructions that project an address out of a guaranteed object and thus create liveness requirements on the borrow scope.

Reply:
fixed